### PR TITLE
Add default ttl for topics

### LINKED ISSUE #163: A default TTL (expiration time) for topics
We could add a default TTL (number and maybe even function) to topic/vault in lib/archivist/index.js, I think that would be very useful for cache lifetime management (configure it once, not for every `set()` call) and it can even replace the responseCache configuration completely.

Thoughts?

### PR DESCRIPTION
**Pull request by Nokitoo**
In lib/archivist/index.js, the user can specify a ttl string (See timeCodeToSec in lib/helpers/helpers.js for format) to expire the topic at a given time.

Resolves #163

### CODE REVIEW

**Comment by Nokitoo** [comment]
As stated by ron in the issue #163, we could also add a function for the ttl. However, I don't see in which specific use case we would need it. Do you have an idea @stelcheck ?

File: lib/archivist/index.js [hunk 1]
```diff
@@ -774,7 +774,13 @@ Archivist.prototype.set = function (topic, index, data, mediaType, encoding, exp
 	trackSize(topic, index, value);
 
 	value.set(mediaType, data, encoding);
-	value.touch(expirationTime);
+
+	if (!expirationTime) {
+		const config = configuration.getTopicConfig(topic);
+		value.touch(Date.now() / 1000 + config.ttl);
```
Comment (stelcheck) [inline lib/archivist/index.js]: What will happen if config.ttl is not set?

Comment (Nokitoo) [reply lib/archivist/index.js]: Right, I will modify it

Comment (Nokitoo) [reply lib/archivist/index.js]: For add, do you mean touch ?

Comment (stelcheck) [reply lib/archivist/index.js]: No, there should be an `archivist.add` normally.

Comment (Nokitoo) [reply lib/archivist/index.js]: I don't see any archivist.add

Comment (stelcheck) [reply lib/archivist/index.js]: Ok, might be some higher level stuff. Don't worry about it.

**Pull request merged**